From 24ba1e244f5bb8af238ff69b70d6b4a12f62b1ef Mon Sep 17 00:00:00 2001
From: "A. Cody Schuffelen" <schuffelen@google.com>
Date: Wed, 6 May 2020 19:41:36 -0700
Subject: [PATCH 10/15] Add libpuresoftkeymasterdevice_host.

This variant removes the keymaster_configuration file, which is only
usable inside Android as it depends on system properties.

Bug: 155328339
Test: m libpuresoftkeymasterdevice_host
Change-Id: If16f0f6bd4fb24da0a689b27948f6f4eb1ca30b0
---
 Android.bp | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/Android.bp b/Android.bp
index fb69928..0cccde3 100644
--- a/Android.bp
+++ b/Android.bp
@@ -213,6 +213,32 @@ cc_library {
     export_include_dirs: ["include"],
 }
 
+cc_library {
+    name: "libpuresoftkeymasterdevice_host",
+    srcs: [
+        "contexts/pure_soft_keymaster_context.cpp",
+        "contexts/soft_keymaster_logger.cpp",
+        "km_openssl/soft_keymaster_enforcement.cpp",
+    ],
+    defaults: ["keymaster_defaults"],
+    host_supported: true,
+    device_supported: false,
+    shared_libs: [
+        "libkeymaster_messages",
+        "libkeymaster_portable",
+        "libsoft_attestation_cert",
+        "liblog",
+        "libcrypto",
+        "libcutils",
+        "libbase",
+    ],
+    clang_cflags: [
+        "-DKEYMASTER_NAME_TAGS",
+        "-fno-rtti", // TODO(b/156427382): Remove workaround when possible.
+    ],
+    export_include_dirs: ["include"],
+}
+
 cc_library_shared {
     name: "libkeymaster3device",
     srcs: [
-- 
2.17.1

